### ISSUE TITLE
Typing issue with 0.4.0

### ISSUE THREAD

**Issue by nartc** [makenotion/notion-sdk-js#200]
Report bugs here only for the Node JavaScript library.

If you're having problems using Notion's API, or have any other feedback about the API including feature requests for the JavaScript library, please email support at developers@notion.com.

**Describe the bug**
Upon upgrading to 0.4.0, typings have gone crazy. `api-types` is now replaced by `api-endpoints`. The problem is the type for each `Block` is now gone. 
**To Reproduce**
Node version: 14
Notion JS library version: 0.4.0

**Expected behavior**
Typing for each Block type should be defined separately and exposed.

**Screenshots**
Like trying to get `HeadingOneBlock` out of this type union is a nightmare
![image](https://user-images.githubusercontent.com/25516557/135121806-a99a1f94-1c49-487b-bc87-eed4d482cb81.png)

![image](https://user-images.githubusercontent.com/25516557/135121507-70e8a0f6-139e-4b47-8d48-07a84c36f5a4.png)

**Additional context**

**Comment by mcginty** [comment]
We've opted to stay with `^0.3` for the time being as well because our types use the built-in notion types (like `RichText`, etc.), which are as far as I'm aware impossible to extract or use given the 0.4 massive "mono-definitions" like `ListBlockChildrenResponse`.

It seems that Notion has automated their type generations, but unfortunately it's not modular enough for those types to be granularly used and referenced elsewhere in the latest release.

Either way, thanks to the Notion team for putting in so much effort into a public API!

**Comment by fzembow** [comment]
You can re-extract the non-root types with tricks like this:

```ts
export type SelectProperty = Extract<
  GetDatabaseResponse["properties"][string],
  { type: "select" }
>;

export type SelectOption = SelectProperty["select"]["options"][number];
```

But I agree, as a new user of this API client, I would expect something like `SelectOption` to be importable rather than have to extract it out.

The TS warnings are really hard to parse out as a result too...

**Comment by huw** [comment]
I argued in the Slack that—regardless of the intention—the change probably encourages best practices, which is to _not_ introspect the types and instead build interfaces for the responses you want. These make your code easier to understand, more modular, and more robust to future API changes.
